bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubWebSocket.java

```
@@ -0,0 +1,238 @@
+ *
+ */
+@NonNullByDefault
+@WebSocket
+public class NeoHubWebSocket extends NeoHubSocketBase {
+
+    private static final int SLEEP_MILLISECONDS = 100;
+    private static final String REQUEST_OUTER = "{\"message_type\":\"hm_get_command_queue\",\"message\":\"%s\"}";
+    private static final String REQUEST_INNER = "{\"token\":\"%s\",\"COMMANDS\":[{\"COMMAND\":\"%s\",\"COMMANDID\":1}]}";
+
+    private final Logger logger = LoggerFactory.getLogger(NeoHubWebSocket.class);
+    private final Gson gson = new Gson();
+    private final WebSocketClient webSocketClient;
+
+    private @Nullable Session session = null;
+    private String responseOuter = "";
+    private boolean responseWaiting;
+
+    /**
+     * DTO to receive and parse the response JSON.
+     *
+     * @author Andrew Fiddian-Green - Initial contribution
+     */
+    private static class Response {
+        @SuppressWarnings("unused")
+        public @Nullable String command_id;
+        @SuppressWarnings("unused")
+        public @Nullable String device_id;
+        public @Nullable String message_type;
+        public @Nullable String response;
+    }
+
+    public NeoHubWebSocket(NeoHubConfiguration config) throws NeoHubException {
+        super(config);
+
+        // initialise and start ssl context factory, http client, web socket client
+        SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
+        sslContextFactory.setTrustAll(true);
+        HttpClient httpClient = new HttpClient(sslContextFactory);
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            throw new NeoHubException(String.format("Error starting http client: '%s'", e.getMessage()));
+        }
+        webSocketClient = new WebSocketClient(httpClient);
+        webSocketClient.setConnectTimeout(config.socketTimeout * 1000);
+        try {
+            webSocketClient.start();
+        } catch (Exception e) {
```

Which exceptions can actually occur from this call? You might not want to catch all exceptions here, including runtime exceptions.

---

Hmm. For me, any exception is an error that needs to be caught. I cannot imagine why I would handle one type of exception differently than another, or in what way.

---

PS the source code of the `webSocketClient.start()` method is declared as 'throws Exception' so the binding code must comply with that declaration :)


```
@Override
public final void start() throws Exception {
...
}
```

---

Sorry, you are right - it's completely fine to catch `Exception` for code out of our control (external API).